### PR TITLE
fix for #1027 display error message when media upload is  disabled

### PR DESCRIPTION
**Pull request by thrijith**
added condition to display an error message when media upload is  disabled

### CODE REVIEW

File: app/assets/js/rtMedia.backbone.js [hunk 1]
```diff
@@ -469,6 +469,14 @@ jQuery( function( $ ) {
 			},
 			initialize: function( config ) {
 				this.uploader = new plupload.Uploader( config );
+				// `ext_enabled` will get value of enabled media types if nothing is enabled,
```
Comment (chandrapatel) [inline app/assets/js/rtMedia.backbone.js]: Don't use ticks. Use single quote.

File: app/assets/js/rtMedia.backbone.js [hunk 2]
```diff
@@ -469,6 +469,14 @@ jQuery( function( $ ) {
...
+				// then an error message will be displayed.
+				var ext_enabled = config.filters[0].extensions.length;
+				if ( ext_enabled === 0 ){
```
Comment (chandrapatel) [inline app/assets/js/rtMedia.backbone.js]: `if ( ext_enabled === 0 ) {`

File: app/assets/js/rtMedia.backbone.js [hunk 3]
```diff
@@ -469,6 +469,14 @@ jQuery( function( $ ) {
 			},
 			initialize: function( config ) {
 				this.uploader = new plupload.Uploader( config );
+				// 'ext_enabled' will get value of enabled media types if nothing is enabled,
+				// then an error message will be displayed.
```
Comment (BhargavBhandari90) [inline app/assets/js/rtMedia.backbone.js]: Use /**/ for multiline comment

File: app/assets/js/rtMedia.backbone.js [hunk 4]
```diff
@@ -839,6 +847,8 @@ jQuery( function( $ ) {
 			} );
 		} else {
 			jQuery( document ).on( 'click', '#rtm_show_upload_ui', function() {
+				// if no media type is enabled error message will be displayed.
```
Comment (BhargavBhandari90) [inline app/assets/js/rtMedia.backbone.js]: First letter should be capital

File: app/main/RTMedia.php [hunk 5]
```diff
@@ -1139,6 +1139,8 @@ function enqueue_scripts_styles() {
 		}
 
 		wp_localize_script( 'rtmedia-backbone', 'rtmedia_upload_progress_error_message', esc_html__( 'There are some uploads in progress. Do you want to cancel them?', 'buddypress-media' ) );
+		// added to display error message when all media types upload are disabled.
```
Comment (BhargavBhandari90) [inline app/main/RTMedia.php]: First letter should be capital

File: app/main/controllers/media/RTMediaComment.php [hunk 6]
```diff
@@ -126,13 +126,16 @@ function remove( $id ) {
...
-		$flag = ( ! (  is_home() || is_post_type_archive() || is_author()))
+		$media_enabled = ( is_rtmedia_upload_music_enabled() || is_rtmedia_upload_photo_enabled()
+			|| is_rtmedia_upload_video_enabled() || is_rtmedia_upload_document_enabled()
+			|| is_rtmedia_upload_other_enabled() );
+		$flag          = ( ! ( is_home() || is_post_type_archive() || is_author() ) )
 		&& is_user_logged_in()
-		&& (is_rtmedia_upload_music_enabled() || is_rtmedia_upload_photo_enabled() || is_rtmedia_upload_video_enabled())
-		 //added condition to disable upload when media is disabled in profile/group but user visits media tab
+		&& ( $media_enabled )
+		// added condition to disable upload when media is disabled in profile/group but user visits media tab.
```
Comment (BhargavBhandari90) [inline app/main/controllers/media/RTMediaComment.php]: First letter should be capital

File: app/main/controllers/shortcodes/RTMediaUploadShortcode.php [hunk 7]
```diff
@@ -33,14 +33,16 @@ public function __construct() {
...
-		$flag = ( ! (  is_home() || is_post_type_archive() || is_author()))
+		$media_enabled = ( is_rtmedia_upload_music_enabled() || is_rtmedia_upload_photo_enabled()
+			|| is_rtmedia_upload_video_enabled() || is_rtmedia_upload_document_enabled()
+			|| is_rtmedia_upload_other_enabled() );
+		$flag          = ( ! ( is_home() || is_post_type_archive() || is_author() ) )
 		&& is_user_logged_in()
-		&& (is_rtmedia_upload_music_enabled() || is_rtmedia_upload_photo_enabled() || is_rtmedia_upload_video_enabled())
-		 //added condition to disable upload when media is disabled in profile/group but user visits media tab
+		&& ( $media_enabled )
+		// added condition to disable upload when media is disabled in profile/group but user visits media tab.
```
Comment (BhargavBhandari90) [inline app/main/controllers/shortcodes/RTMediaUploadShortcode.php]: First letter should be capital

**Pull request merged**